msccl/language/instruction_dag.py

```diff
@@ -144,10 +144,16 @@ def _read(self, rank, buffer, index, size, op):
             op.prev.add(prev_op)
 
     def _infer_dependencies(self):
+        visited = set()
         for slot, ops in self.operations.items():
+            if ops in visited:
```

Comment:
To my understanding, "ops" is still a single op instead of multiple "op"s? Shall we change all "ops" to "op" to avoid confusion?

Reply:
Thanks, fixed